packages/opto-wallet/src/lib/opto-wallet.ts

```diff
@@ -0,0 +1,255 @@
+  });
+
+  const wallet = new WalletConnection(near, "near_app");
+
+  return {
+    wallet,
+    keyStore,
+  };
+};
+
+const OptoWallet: WalletBehaviourFactory<
+  BrowserWallet,
+  { params: OptoWalletExtraOptions }
+> = async ({ metadata, options, store, params, logger }) => {
+  const _state = await setupWalletState(params, options.network);
+
+  const getAccounts = () => {
+    const accountId: string | null = _state.wallet.getAccountId();
+
+    if (!accountId) {
+      return [];
+    }
+
+    return [{ accountId }];
+  };
+
+  const transformTransactions = async (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    const account = _state.wallet.account();
+    const { networkId, signer, provider } = account.connection;
+
+    const localKey = await signer.getPublicKey(account.accountId, networkId);
+
+    return Promise.all(
+      transactions.map(async (transaction, index) => {
+        const actions = transaction.actions.map((action) =>
+          createAction(action)
+        );
+        const accessKey = await account.accessKeyForTransaction(
+          transaction.receiverId,
+          actions,
+          localKey
+        );
+
+        if (!accessKey) {
+          throw new Error(
+            `Failed to find matching key for transaction sent to ${transaction.receiverId}`
+          );
+        }
+
+        const block = await provider.block({ finality: "final" });
+
+        return nearTransactions.createTransaction(
+          account.accountId,
+          utils.PublicKey.from(accessKey.public_key),
+          transaction.receiverId,
+          accessKey.access_key.nonce + index + 1,
+          actions,
+          utils.serialize.base_decode(block.header.hash)
+        );
+      })
+    );
+  };
+
+  return {
+    async signIn({ contractId, methodNames }) {
+      const existingAccounts = getAccounts();
+
+      if (existingAccounts.length) {
+        return existingAccounts;
+      }
+
+      await _state.wallet.requestSignIn({ contractId, methodNames });
+
+      return getAccounts();
+    },
+
+    async signOut() {
+      if (_state.wallet.isSignedIn()) {
+        _state.wallet.signOut();
+      }
+    },
+
+    async getAccounts() {
+      return getAccounts();
+    },
+
+    async verifyOwner({ message, callbackUrl, meta }) {
+      logger.log("verifyOwner", { message });
+
+      const account = _state.wallet.account();
+
+      if (!account) {
+        throw new Error("Wallet not signed in");
+      }
+      const locationUrl =
+        typeof window !== "undefined" ? window.location.href : "";
+
+      const url = callbackUrl || locationUrl;
+
+      if (!url) {
+        throw new Error(`The callbackUrl is missing for ${metadata.name}`);
+      }
+
+      const encodedUrl = encodeURIComponent(url);
+      const extraMeta = meta ? `&meta=${meta}` : "";
+
+      window.location.replace(
+        `${params.walletUrl}/verify-owner?message=${message}&callbackUrl=${encodedUrl}${extraMeta}`
+      );
```

Comment:
Looks like you have not implemented the `verifyOwner` functionality in your wallet. The same goes for the WalletConnect support you have.
This is a new feature that's been requested from Wallet Selector and we don't have a NEP/SPEC for this yet but there are already some implementations I think this info will be helpful:

Here's an implementation for MyNearWallet (browser wallet): 
https://github.com/near/near-wallet/pull/2801

Here is our implementation for WalletConnect (web-examples): 
https://github.com/WalletConnect/web-examples/pull/59
https://github.com/WalletConnect/web-examples/pull/59/commits/25b1a4c86667eb81cbf78eace0fc2bf80e1356b1

Reply:
Just to clarify this functionality is not a blocker for merging this PR.
If it's necessary you can open a new PR once you're ready.